Source/DafnyCore/Compilers/SinglePassCompiler.cs

```diff
@@ -4331,7 +4331,9 @@ private class GetterSetterLvalueImpl : ILvalue {
         //   <prelude>   // filled via copyInstrWriters -- copies out-parameters used in letexpr to local variables
         //   ss          // translation of ss has side effect of filling the top copyInstrWriters
         var w = writer;
-        if (ss.Labels != null) {
+        if (ss.Labels != null && !(ss is VarDeclPattern or VarDeclStmt)) {
```

Comment:
Definitely sounds like a good change, but how confident are we that no other statement cases would similarly be broken by being wrapped in a block?

Reply:
The only problem of wrapping with block quotes (at least in javascript), is that it creates a "local scope". Other than that, you can insert {} around any statement without changing its meaning.
Hence, in our case, the only place where this matter is that we are introducing new variables for subsequent statements. All other statements are only modifying existing vars that were already in scope, so they have no reason to be broken by being wrapped into a labelled block.